File: go/samples/send_data/send_data.go
package main

import (
	"fmt"

	"github.com/bloock/bloock-sdk-go/v2"
	"github.com/bloock/bloock-sdk-go/v2/builder"
	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/client/entity"
	"github.com/bloock/bloock-sdk-quickstart/utils"
	"github.com/bloock/bloock-sdk-quickstart/utils/logger"
)

func main() {
	utils.Sample("Send Data", func(c utils.Config) error {
		// we set the API key and create a client
		bloock.ApiKey = c.ApiKey
		sdk := client.NewClient()

		// we create an array of records which will contain the records we want to send
		records := []entity.Record{}

		// first we create a record
		record, err := builder.NewRecordBuilderFromString("Hello world").Build()
		if err != nil {
			return err
		}

		// we then get its hash
		hash, err := record.GetHash()
		if err != nil {
			return err
		}

		logger.Success("Hash: " + hash)

		// append the record we want to send to the array
		records = append(records, record)

		// finally we can send the records
		receipt, err := sdk.SendRecords(records)
		if err != nil {
			return err
		}
		// we get a receipt with informationa about the transaction
		logger.Success(fmt.Sprintf("Record receipts: %+v", receipt))

		return nil
	})
}

File: java/src/main/java/com/bloock/quickstart/samples/send_data/SendData.java
package com.bloock.quickstart.samples.send_data;

import com.bloock.quickstart.utils.Config;
import com.bloock.quickstart.utils.Logger;
import com.bloock.quickstart.utils.Sample;
import com.bloock.sdk.Bloock;
import com.bloock.sdk.builder.Builder;
import com.bloock.sdk.client.Client;
import com.bloock.sdk.entity.Record;
import com.bloock.sdk.entity.RecordReceipt;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class SendData extends Sample {
  public void run(Config config) throws Exception {
    Bloock.apiKey = config.apiKey;
    Client client = new Client();

    // we create an ArrayList of strings which will contain
    // the hashes of the records we want to send
    Record record = Builder.fromString("Hello world").build();
    String hash = record.getHash();
    ArrayList<String> records = new ArrayList<>(Arrays.asList(hash));

    // finally we can send the records
    List<RecordReceipt> receipts = client.sendRecords(records);

    // we get a receipt with informationa about the transaction
    Logger.success("Record receipts: ");
    receipts.stream()
        .forEach(
            x -> {
              Logger.success(
                  "anchor: "
                      + x.getAnchor()
                      + ", client: "
                      + x.getClient()
                      + ", record: "
                      + x.getRecord()
                      + ", status: "
                      + x.getStatus());
            });
  }

  SendData(String name) {
    super(name);
  }

  public static void main(String[] args) {
    new SendData("SendData");
  }
}
